### PR TITLE
Add GaugeAdderMigrationController

### PR DESCRIPTION
**Pull request by TomAFrench**
This PR adds the GaugeAdderMigrationController.

### CODE REVIEW

File: pkg/governance-scripts/contracts/20220628-gauge-adder-migration/GaugeAdderMigrationCoordinator.sol [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    function _afterLastStage() internal virtual override {
+        ICurrentAuthorizer authorizer = ICurrentAuthorizer(address(getAuthorizer()));
+
+        authorizer.renounceRole(authorizer.DEFAULT_ADMIN_ROLE(), address(this));
+    }
+
+    // Internal functions
+
+    function _setupOptimismGaugeType() private {
```
Comment (TomAFrench) [inline pkg/governance-scripts/contracts/20220628-gauge-adder-migration/GaugeAdderMigrationCoordinator.sol:66]: https://github.com/balancer-labs/balancer-v2-monorepo/pull/1432#discussion_r912081799

File: pkg/governance-scripts/contracts/20220628-gauge-adder-migration/GaugeAdderMigrationCoordinator.sol [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        ICurrentAuthorizer authorizer = ICurrentAuthorizer(address(getAuthorizer()));
+
+        bytes32 addGaugeTypeRole = getAuthorizerAdaptor().getActionId(IGaugeController.add_type.selector);
+        authorizer.grantRole(addGaugeTypeRole, address(this));
```
[the remark this answers is not in the document]
Comment (TomAFrench) [reply pkg/governance-scripts/contracts/20220628-gauge-adder-migration/GaugeAdderMigrationCoordinator.sol:70]: That's true, I'll have a look into how to do this in a clean way (adaptor causes a UX hurdle) another time.

File: pkg/governance-scripts/contracts/20220628-gauge-adder-migration/GaugeAdderMigrationCoordinator.sol [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+        // Create "Optimism" gauge type on GaugeController.
+        // All types on the Gauge controller have equal type weights of 1e18.
+        uint256 typeWeight = 1e18;
+        getAuthorizerAdaptor().performAction(
+            address(gaugeController),
+            abi.encodeWithSelector(IGaugeController.add_type.selector, "Optimism", typeWeight)
+        );
```
[the remark this answers is not in the document]
Comment (TomAFrench) [reply pkg/governance-scripts/contracts/20220628-gauge-adder-migration/GaugeAdderMigrationCoordinator.sol:78]: We could do however I aimed to keep this as focused as possible. There's also some uncertainty around Gnosis and zksync so I thought it was worth waiting a little bit to take an irreversible action.

File: pkg/deployments/tasks/scripts/20220629-gauge-adder-migration-coordinator/input.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+const AuthorizerAdaptor = new Task('20220325-authorizer-adaptor', TaskMode.READ_ONLY);
+const OldGaugeAdder = new Task('20220325-gauge-adder', TaskMode.READ_ONLY);
+const NewGaugeAdder = new Task('20220628-gauge-adder-v2', TaskMode.READ_ONLY);
+
+const ArbitrumRootGaugeFactory = new Task('20220413-arbitrum-root-gauge-factory', TaskMode.READ_ONLY);
+const OptimismRootGaugeFactory = new Task('20220628-optimism-root-gauge-factory', TaskMode.READ_ONLY);
+
+// Placeholders
+const LiquidityMiningMultisig = '0x0000000000000000000000000000000000000420';
+const GaugeCheckpointingMultisig = '0x0000000000000000000000000000000000000421';
```
Comment (nventuro) [inline pkg/deployments/tasks/scripts/20220629-gauge-adder-migration-coordinator/input.ts:22]: Can we fill these in?

File: pkg/governance-scripts/contracts/20220628-gauge-adder-migration/GaugeAdderMigrationCoordinator.sol [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+                IArbitrumFeeProvider.setArbitrumFees.selector
+            ),
+            gaugeCheckpointingMultisig
+        );
+        authorizer.grantRole(
+            IAuthentication(address(optimismRootGaugeFactory)).getActionId(
+                IOptimismGasLimitProvider.setOptimismGasLimit.selector
+            ),
+            gaugeCheckpointingMultisig
+        );
```
Comment (nventuro) [inline pkg/governance-scripts/contracts/20220628-gauge-adder-migration/GaugeAdderMigrationCoordinator.sol:89]: Let's add tests for these two permissions.

**Review by nventuro — COMMENTED** [review]
Great! Looks like we're just missing a simple test and finalizing the inputs

**Review by nventuro — APPROVED** [review]
Looks ready to merge and deploy! Great work :tada:

**Pull request merged**